src/java/org/apache/cassandra/db/memtable/Memtable_API.md

```diff
@@ -0,0 +1,161 @@
+# Memtable API
```

Comment:
This file is missing the Apache license header. Also, maybe we want to place this file into the `doc` dir?

Reply:
Fixed the license check as done for `Paxos.md`. Developer docs make more sense with the code that they describe; we could split this into user-facing and developer-facing parts, but I'd rather have user-facing docs only when we use this in force with trie/pmem memtables.